testapi.pm

```diff
@@ -121,6 +121,8 @@ sub init {
     }
     elsif (get_var('SERIALDEV')) {
         $serialdev = get_var('SERIALDEV');
+        set_var('SERIALDEVORIG', $serialdev);
+        bmwqemu::save_vars();
```

Comment:
I don't think this is a good idea, also see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6800/files#r257496910 please:

1. The backend should not need to save variables which are only relevant in test context. This needs to be handled in tests
2. When coming up with new variable names better use a word separation character, e.g. `_` so `SERIAL_DEV_ORIG` or at least `SERIALDEV_ORIG`
3. Define all new variables in the documentation, the file https://github.com/os-autoinst/os-autoinst/blob/master/doc/backend_vars.asciidoc

But as stated in 1. I do not think this should be here at all and the PR should be closed.

Reply:
As stated in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6800#discussion_r257538430
The original SERIALDEV is overwritten, at least, at one place, for example:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/a5371a62e9829efd54b547160111c2699da61900/lib/Utils/Backends.pm#L50
It is not a good design and good practice in the first place. And because of this bad design, I have to save its original value again to another parameter.  Why didn't you prevent this bad design from happening in the first place ?  @okurz

Reply:
> Why didn't you prevent this bad design from happening in the first place ? @okurz

:D Because I am not almighty ;)

Reply:
@okurz The most suitable solution coming to my mind currently is introducing a new parameter called SERIALDEV_AARCH64 or similar or more meaningful. We can not solve this problem all at once, because the original SERIALDEV is already being used for multiple purposes(at least, sshserial and serial console at the same time).  If the original implementation got refined more carefully, there should have been two parameters already, one for sshserial and one for serial console, so I do not need to bother add another one by myself. But in light of the existing fact, and  we do need one more to continue our work, my proposal is keep the original implementation and add another one SERIALDEV_AARCH64 for aarch64 machine serial console usage. Then this PR can be closed, and the other one can be updated with this new proposal. Sounds reasonable ?

Reply:
I do not see where aarch64 is special here. I would rather use a dedicated variable for the serial-/ssh-terminal

Reply:
Some aarch64 use ttyAMA0, some aarch64 use ttyS0, and these machines can not be configured to use other serial consoles. They only have one to use. I will see what I can do. So you suggest one parameter for serial terminal, one for ssh terminal, right ?

Reply:
I agree with okurz. SERIALDEV is better to be set or changed or saved in test code, just as the way we do for xen serial console setting on x86 testing.

Reply:
@okurz  I am really willing to hear from you with your opinion. Which of the following suggestions that you support:
1. Introduce another PARAMETER for AARCH64 serial console
2. Change the existing implementation, which means do not save 'sshserial' to SERIALDEV, save it to local variable or save it to new SSHSERIALDEV parameter
3. Your own proposal

Reply:
@alice-suse Can you point me on how do you do it with XEN? I see the point of @waynechen55 about different consoles for aarch64, thus should be a variable that changes where the console is.

Since IPMI workers are usually set as jump hosts, we can easily put that into a new variable in the workers.ini file. Unless there's a difference in the worker setup that I'm unable to see?

Could you guys provide a bit more context? @alice-suse @waynechen55

Reply:
@foursixnine For the worker setup, there is no difference. I am really curious about and interested in your "easily put that into a new variable in the workers.ini file". Could you help show me an example ? The workers.ini file and how to use it. 
And you said "Since IPMI workers are usually set as jump hosts", are there any prerequisites for setting up such a jump host ? Hardware and software requirements ? Let's say there are totally 6 openqa-workers need to be established.

Reply:
Regarding https://github.com/os-autoinst/os-autoinst/pull/1107#discussion_r258373931 I would go with your proposal 2.

Reply:
Already made the change based on proposal 2.